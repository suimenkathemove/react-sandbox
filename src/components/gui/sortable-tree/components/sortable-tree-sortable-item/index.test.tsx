import 'jest-styled-components';

import { cleanup, render } from '@testing-library/react';

import { defaultProps } from './index.stories';

import { SortableTreeSortableItem } from './';

const setup = () => render(<SortableTreeSortableItem {...defaultProps} />);

describe('SortableTreeSortableItem', () => {
  afterEach(() => {
    cleanup();
  });

  test('render', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
