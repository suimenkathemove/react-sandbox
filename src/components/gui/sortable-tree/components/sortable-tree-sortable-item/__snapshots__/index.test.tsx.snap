// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableTreeSortableItem render 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 16px;
  height: 24px;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  border-radius: 4px;
}

.c2 svg {
  width: 0.5em;
  height: 0.5em;
  fill: #757575;
}

.c2:hover {
  background-color: rgba(0,0,0,0.05);
}

.c2:hover svg {
  fill: #6f7b88;
}

.c2:active {
  background-color: inherit;
}

.c2:active svg {
  fill: inherit;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 16px;
  height: 24px;
  border-radius: 4px;
}

.c5:hover {
  background-color: rgba(0,0,0,0.05);
}

.c5:hover svg {
  fill: #6f7b88;
}

.c0 {
  padding: 0 0 0 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 460px;
  height: 40px;
  padding: 0 0.5rem;
  background-color: #ffffff;
  border: solid 1px #e0e0e0;
  border-radius: 4px;
}

.c3 {
  margin-left: 0.5rem;
  font-family: 'Hiragino Kaku Gothic Pro',sans-serif;
  font-size: 0.75rem;
  font-weight: bold;
  color: #424242;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
}

.c4 {
  margin-left: auto;
}

<li
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        aria-describedby=""
        aria-roledescription="sortable"
        class="c2"
        role="button"
        tabindex="0"
      >
        <svg
          viewBox="0 0 20 20"
          width="12"
        >
          <path
            d="M7 2a2 2 0 1 0 .001 4.001A2 2 0 0 0 7 2zm0 6a2 2 0 1 0 .001 4.001A2 2 0 0 0 7 8zm0 6a2 2 0 1 0 .001 4.001A2 2 0 0 0 7 14zm6-8a2 2 0 1 0-.001-4.001A2 2 0 0 0 13 6zm0 2a2 2 0 1 0 .001 4.001A2 2 0 0 0 13 8zm0 6a2 2 0 1 0 .001 4.001A2 2 0 0 0 13 14z"
          />
        </svg>
      </button>
      <span
        class="c3"
      >
        1
      </span>
      <div
        class="c4"
      >
        <button
          class="c5"
        >
          <svg
            viewBox="0 0 22 22"
            width="8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.99998 -0.000206962C2.7441 -0.000206962 2.48794 0.0972617 2.29294 0.292762L0.292945 2.29276C-0.0980552 2.68376 -0.0980552 3.31682 0.292945 3.70682L7.58591 10.9998L0.292945 18.2928C-0.0980552 18.6838 -0.0980552 19.3168 0.292945 19.7068L2.29294 21.7068C2.68394 22.0978 3.31701 22.0978 3.70701 21.7068L11 14.4139L18.2929 21.7068C18.6829 22.0978 19.317 22.0978 19.707 21.7068L21.707 19.7068C22.098 19.3158 22.098 18.6828 21.707 18.2928L14.414 10.9998L21.707 3.70682C22.098 3.31682 22.098 2.68276 21.707 2.29276L19.707 0.292762C19.316 -0.0982383 18.6829 -0.0982383 18.2929 0.292762L11 7.58573L3.70701 0.292762C3.51151 0.0972617 3.25585 -0.000206962 2.99998 -0.000206962Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
